arroyo-api/src/optimizations.rs

```diff
@@ -44,39 +44,43 @@ fn remove_in_place(graph: &mut DiGraph<StreamNode, StreamEdge>, node: NodeIndex)
 
 fn fuse_window_aggregation(graph: &mut DiGraph<StreamNode, StreamEdge>) {
     'outer: loop {
-        let sources: Vec<_> = graph.externals(Direction::Incoming).collect();
+        let sources: Vec<_> = graph.externals(Incoming).collect();
 
         for source in sources {
             let mut dfs = Dfs::new(&(*graph), source);
 
             while let Some(idx) = dfs.next(&(*graph)) {
                 let operator = graph.node_weight(idx).unwrap().operator.clone();
-                let mut ins = graph.edges_directed(idx, Direction::Incoming);
+                let mut ins = graph.edges_directed(idx, Incoming);
 
                 let in_degree = ins.clone().count();
                 let no_shuffles = ins.all(|e| e.weight().typ == EdgeType::Forward);
 
-                let mut ins = graph.edges_directed(idx, Direction::Incoming);
-                if no_shuffles && in_degree == 1 {
-                    let source_idx = ins.next().unwrap().source();
-                    let in_node = graph.node_weight_mut(source_idx).unwrap();
-                    if let Operator::Window { agg, .. } = &mut in_node.operator {
-                        if agg.is_none() {
-                            let new_agg = match &operator {
-                                Operator::Count => Some(WindowAgg::Count),
-                                Operator::Aggregate(AggregateBehavior::Min) => Some(WindowAgg::Min),
-                                Operator::Aggregate(AggregateBehavior::Max) => Some(WindowAgg::Max),
-                                Operator::Aggregate(AggregateBehavior::Sum) => Some(WindowAgg::Sum),
-                                _ => None,
-                            };
-
-                            if let Some(new_agg) = new_agg {
-                                *agg = Some(new_agg);
-                                remove_in_place(graph, idx);
-                                // restart the loop if we change something
-                                continue 'outer;
-                            }
-                        }
+                let mut ins = graph.edges_directed(idx, Incoming);
+                if !(no_shuffles && in_degree == 1) {
+                    continue;
+                }
+
+                let source_idx = ins.next().unwrap().source();
+                let in_node = graph.node_weight_mut(source_idx).unwrap();
+                if let Operator::Window { agg, .. } = &mut in_node.operator {
+                    if !agg.is_none() {
```

Comment:
It might be cleaner as 
```suggestion
                    if agg.is_some() {
```

Reply:
That's a good idea.